samcli/commands/_utils/options.py

```diff
@@ -162,6 +162,22 @@ def parameter_override_option(f):
     return parameter_override_click_option()(f)
 
 
+def no_progressbar_click_option():
+    return click.option(
+        "--no-progressbar",
+        default=False,
+        required=False,
+        is_flag=True,
+        help="Indicates whether to simplify the logging "
```

Comment:
help section does need to go into implementation details. it can simply say

```
Does not showcase a progress bar when uploading artifacts to s3
```